File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.java
class Solution {
    public int[] replaceElements(int[] arr) {
        int max;
        for(int i = 0; i < arr.length-1; i++){
            max = -1;
            for(int j = i+1; j < arr.length; j++){
                if(arr[j] > max)
                    max = arr[j];
            }
            arr[i] = max;
        }
        arr[arr.length-1] = -1;     //for the last element
        return arr;
    }
}

/*
#Another Solution#

class Solution {
    public int[] replaceElements(int[] arr) {

        int max = -1;

        for (int i = arr.length - 1; i >= 0; i--) {
            int temp = arr[i];
            arr[i] = max;
            max = Math.max(temp, max);
        }

        return arr;
    }
}
*/

File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.swift
class Solution {
    func replaceElements(_ arr: [Int]) -> [Int] {
        var res = [Int]()
        
        var i = 0
        while i < arr.count-1 {
            var j = i+1
            var max = -1
            while j < arr.count {
                if arr[j] > max { max = arr[j] }
                j += 1
            }
            
            res.append(max)
            i += 1
        }
        res.append(-1)    // for the last element
        
        return res
    }
}